### PR TITLE
gigamon: Update package description and cleanup

### PR DESCRIPTION
**Pull request by kcreddy**
## Proposed commit message
Update package description and readme cleanup.

Current package refers to be ingesting data from `Filebeat`. 
Updated this to `Elastic Agent` inside the package description.

Other Cleanup:
 - Fix comments positioning in ingest pipeline.
 - Fix `README` indentation.
 - Fix dashboards containing undefined index-pattern.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Screenshots

After `README` update, the properties are indented properly:
Screenshot 2024-08-26 at 4 36 08 PM

### CODE REVIEW

File: packages/gigamon/manifest.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 format_version: 3.1.3
 name: gigamon
 title: Gigamon
-version: "0.1.0"
-description: "This Gigamon package integrates with Filebeat to collect and ingest data from Gigamon devices"
+version: "0.2.0"
+description: This Gigamon package integrates with Elastic Agent to collect and ingest data from Gigamon devices.
```
Comment (andrewkroh) [inline packages/gigamon/manifest.yml:5]: "Collect X from Y with Elastic Agent." should be the format.

File: packages/gigamon/data_stream/ami/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -831,11 +831,6 @@ processors:
         - json
       if: ctx.tags == null || !(ctx.tags.contains('preserve_duplicate_custom_fields'))
       ignore_missing: true
-  - remove:
-      field: event.original
-      if: ctx.tags == null || !(ctx.tags.contains('preserve_original_event'))
-      ignore_failure: true
-      ignore_missing: true
```
Comment (efd6) [inline packages/gigamon/data_stream/ami/elasticsearch/ingest_pipeline/default.yml:838]: Why is this removed?

Comment (kcreddy) [reply packages/gigamon/data_stream/ami/elasticsearch/ingest_pipeline/default.yml:838]: I thought since we are handling the `preserve_original_event` inside the fleet final pipeline, this is no longer needed. WDYT?

Comment (kcreddy) [reply packages/gigamon/data_stream/ami/elasticsearch/ingest_pipeline/default.yml:838]: It is handled here: https://github.com/kpollich/kibana/blob/main/x-pack/plugins/fleet/server/constants/fleet_es_assets.ts#L126-L131

Comment (efd6) [reply packages/gigamon/data_stream/ami/elasticsearch/ingest_pipeline/default.yml:838]: Yeah, fair enough.

**Review by efd6 — APPROVED** [review]
